### DISCUSSION TITLE
Connect with MongoDB

### DISCUSSION THREAD

**Discussion by teomikitenko** [prisma/prisma#23915] [Q&A]
### Question

Hello. Im using npx prisma db pull command and strange dns issue Error: MongoDB error
Kind: An error occurred during DNS resolution: proto error: io error: A socket operation was attempted to an unreachable network. (os error 10051), labels: {}

### How to reproduce (optional)

npx db pull

### Expected behavior (optional)
I just wanted take my model from db and test prisma

### Information about Prisma Schema, Client Queries and Environment (optional)

```datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}
```

- OS:  Windows 10
- Database: MongoDB- Node.js version: v21.6.0

Prisma : latest version

**Comment by ludralph** [comment]
Hi @teomikitenko 👋 

Thank you for raising this question.

This looks like an issue with your internet connection. Are you using a VPN? Can you connect to the MongoDB instance using other tools like [Compass](https://www.mongodb.com/products/tools/compass)? 

I was able to find a related discussion [here](https://github.com/prisma/prisma/discussions/14908) where other users posted what worked for them. You may want to check it out as well.

**Reply by teomikitenko** [reply]
Yes im reading this discussion but i dont use VPN, Compass  connection with Mongo being success

**Reply by teomikitenko** [reply]
hmm but if i use cloudflare vpn all start working. how fix this behevior - need working without vpn

**Reply by ludralph** [reply]
Hi @teomikitenko 

To fix this behavior and make it work without a VPN, you can try the following solutions:

a. Check your /etc/resolv.conf file: As mentioned in [this GitHub issue comment](https://github.com/prisma/prisma/issues/13370), there might be an issue with your DNS configuration. Look for any malformed entries in your /etc/resolv.conf file, particularly lines containing search -- or similar. If you find such entries, you may need to comment them out or remove them.

b. Use a different DNS server: Try changing your DNS server to a public DNS provider like Google (8.8.8.8) or Cloudflare (1.1.1.1). This might help bypass any DNS resolution issues with your current setup.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
